### PR TITLE
Remove deprecated RenderObjectElement methods

### PR DESCRIPTION
**Pull request by Piinks**
This removes the following methods from `RenderObjectElement` 
- `insertChildRenderObject`
- `moveChildRenderObject`
- `removeChildRenderObject`

These deprecated methods have reached end of life after the release of Flutter 2.10

- The respective replacements are
  - `insertRenderObjectChild`
  - `moveRenderObjectChild`
  - `removeRenderObjectChild`
- ✅  This migration is supported by the dart fix tool. 
- 🕐  This API was deprecated in https://github.com/flutter/flutter/pull/64254, first tagged v1.21

Part of https://github.com/flutter/flutter/issues/98537
For the full list of deprecations being removed in this batch, see [flutter.dev/go/deprecations-removed-after-2-10](https://docs.google.com/spreadsheets/d/12krawYCu6X_g_5wLGpmiAIi-VcTRV6xG7RGbRovuatQ/edit?usp=sharing)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/framework.dart [hunk 1]
```diff
@@ -6026,35 +6009,22 @@ abstract class RenderObjectElement extends Element {
         ),
         ErrorDescription(
           'insertChildRenderObject() has been deprecated in favor of '
-          'insertRenderObjectChild(). See https://github.com/flutter/flutter/issues/63269 '
-          'for details.',
+              'insertRenderObjectChild(). See https://github.com/flutter/flutter/issues/63269 '
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/framework.dart:6012]: nit: revert these format changes?

**Comment by Piinks** [comment]
@goderbauer can you review this one again? It should be ready to go now. :)

**Review by goderbauer — APPROVED** [review]
LGTM

Thank you sooo much for cleaning this up!

**Comment by Piinks** [comment]
There are a few more references that came up, still migrating.

**Comment by Piinks** [comment]
Ok, now we should be ready here.

**Pull request merged**